src/pg_to_evalscript/javascript_datacube/DataCube.js

```diff
@@ -85,6 +96,27 @@ class DataCube {
         })
     }
 
+    extendDimensionWithData(dimension, dataToAdd) {
+        const finalShape = this.getDataShape()
+        finalShape[dimension]++;
+        
+        const finalData = this.insertIntoDimension(dataToAdd, finalShape[dimension], finalShape[dimension] - 1)     
+        this.data = ndarray(finalData, finalShape)
+    }
+
+    insertIntoDimension(dataToAdd, dimensionSize, locationInDimension) {
```

Comment:
I don't think this works correctly generally - if I'm not mistaken. Eg. if you would have 2 dimensions `t` and `bands`, and tried inserting in `t`, it would insert in the same position, just set a different shape:
```js
d = Datacube([1,2,3,4], shape=(2,2), dims=(t,band))
d. extendDimensionWithData(1, [9,99])
=> Datacube([1,2,9,3,4,99], shape=(2,3), dims=(t,band)) // correct

d = Datacube([1,2,3,4], shape=(2,2), dims=(t,band))
d. extendDimensionWithData(0, [9,99])
=> Datacube([1,2,9,3,4,99], shape=(3,2), dims=(t,band)) // incorrect, looks like [[1,2],[9,3],[4,99]], should be [[1,2],[3,4],[9,99]]
```
But we can fix that in a separate PR. Meanwhile we can just rename these functions to `extendFinalDimensionWithData`/`insertIntoFinalDimension` or smth like that. It's not quite correct, as the `bands` dimension is not _necessarily_ the last one, but it will do for now.

Reply:
Renamed the functions to the suggested names.